### ISSUE TITLE
Input not being evaluated against policy

### ISSUE THREAD

**Issue by migscabral** [open-policy-agent/opa#4455]
## Short description 

TLDR; 
- Input is not being evaluated against policy, even if data and policy belongs to the same namespace.
- I can only access my policy by listing all policies using `/v1/policies`, and I'm not able to access through `/v1/policies/<policy id>`.
- I'm not sure if I'm referencing data in my rego code correctly.

Seems like the input is only being evaluated against the data and not together with the policy. From this [example](https://www.openpolicyagent.org/docs/v0.12.2/rest-api/#example-request-5), it seems like the correct way of using the Data API is to use the namespace + method. In my case it should be `/v1/data/actions/allow`. I tried making a request to that endpoint but it always returns a response containing only the decision id. 

Sample call:
```
curl --location --request POST 'http://localhost:8282/v1/data/actions/allow' \
--header 'Content-Type: application/json' \
--data-raw '{
    "action": "action-1",
    "action_group": "action-group-1",
    "role": "role-1"
}'
```

Sample response:
```
{
    "decision_id": "564a2d8b-1b14-46ca-826d-a72afdd46a1b"
}
```

I also tried making a request to `/v1/data/actions` and it also returns a response containing only the decision id.

I ran `inspect` on the bundle directory and it seems like the files are properly recognized and namespaced:
```
docker run -v ${PWD}/policies:/policies openpolicyagent/opa:0.38.0 inspect /policies
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
MANIFEST:
+-------+---------------+
| FIELD |     VALUE     |
+-------+---------------+
| Roots | authz/actions |
|       | authz/urls    |
+-------+---------------+
NAMESPACES:
+--------------------+-------------------------------------+
|     NAMESPACE      |                FILE                 |
+--------------------+-------------------------------------+
| data.authz.actions | /policies/authz/actions/policy.rego |
|                    | /policies/authz/actions/data.json   |
| data.authz.urls    | /policies/authz/urls/policy.rego    |
|                    | /policies/authz/urls/data.json      |
+--------------------+-------------------------------------+
```

I also tried listing my policies using `/v1/policies` to check if OPA sees them and it seems like they are loaded (trimmed the output to fit in this ticket):
```
{
    "result": [
        {
            "id": "/policies/policies/authz/urls/policy.rego",
            "raw": "package authz.urls\n\nimport future.keywords.in\n\ndefault allow = false\n\nallow {\n\tinput.role == \"admin\"\n}\n\nallow {\n\tsome grant\n\tuser_is_granted[grant]\n\n\tinput.url == grant\n}\n\nuser_is_granted[grant] {\n\tsome i\n    data.authz.urls.role_url_resources[i].role == input.role\n    some grant in data.authz.urls.role_url_resources[i].urls\n}\n",
            "ast": {
                "package": {
                    "path": [
                        {
                            "type": "var",
                            "value": "data"
                        },
                        {
                            "type": "string",
                            "value": "authz"
                        },
                        {
                            "type": "string",
                            "value": "urls"
                        }
                    ]
                },
                "rules": [
                    ...
                ]
            }
        },
        {
            "id": "/policies/policies/authz/actions/policy.rego",
            "raw": "package authz.actions\n\nimport future.keywords.in\n\ndefault allow = false\n\nallow {\n\tinput.role == \"admin\"\n}\n\nallow {\n\tsome grant\n\tuser_is_granted[grant]\n\n\tinput.action == grant\n}\n\nuser_is_granted[grant] {\n    some i\n\tdata.authz.actions.role_actions[i].role == input.role\n    actions := data.authz.actions.role_actions[i][\"actions\"][input.action_group]\n\tsome grant in actions\n}\n",
            "ast": {
                "package": {
                    "path": [
                        {
                            "type": "var",
                            "value": "data"
                        },
                        {
                            "type": "string",
                            "value": "authz"
                        },
                        {
                            "type": "string",
                            "value": "actions"
                        }
                    ]
                },
                "rules": [
                    ...
                ]
            }
        }
    ]
}
```

However I can't access each individual policy. I tried the following:
- `/v1/policies/policies/policies/authz/urls/policy.rego`
- `/v1/policies/policies/policies/authz/urls/policy`
- `/v1/policies/policies/policies/authz/urls`
- `/v1/policies/policies/authz/urls/policy.rego`
- `/v1/policies/policies/authz/urls/policy`
- `/v1/policies/policies/authz/urls`
- `/v1/policies/authz/urls/policy.rego`
- `/v1/policies/authz/urls/policy`
- `/v1/policies/authz/urls`

All returned the same error response:
```
{
    "code": "resource_not_found",
    "message": "storage_not_found_error: policy id <url>"
}
```

In my code I reference the data this way since I assumed data is in the specified namespace returned by `inspect`:
```
...
    data.authz.actions.role_actions[i].role == input.role
    actions := data.authz.actions.role_actions[i]["actions"][input.action_group]
...
```

**OPA version:** 0.38.1 running on Docker v20.10.12
-->

## Steps To Reproduce
1. Unzip the project directory I attached. Navigate to the project directory.
2. Execute `docker-compose up` 
3. Execute the following requests:
```
curl --location --request POST 'http://localhost:8282/v1/data/actions/allow' \
--header 'Content-Type: application/json' \
--data-raw '{
    "action": "action",
    "action_group": "action-group-1",
    "role": "role-1"
}'
```

```
curl --location --request POST 'http://localhost:8282/v1/data/urls/allow' \
--header 'Content-Type: application/json' \
--data-raw '{
    "role": "role-1",
    "url": "/allowed/url/1"
}'
```

4. Additional step: inspect the bundle directory
```
docker run -v ${PWD}/policies:/policies openpolicyagent/opa:0.38.0 inspect /policies
```

## Expected behavior
When calling Data API, input should be evaluated against both data and policy if the data and policy belong to the same namespace.

## Additional Context
- See attached project directory: [opa.zip](https://github.com/open-policy-agent/opa/files/8319845/opa.zip)

**Comment by ashutosh-narkar** [comment]
To evaluate a policy with some input you would do something like:

```
 curl -X POST -H "Content-Type: application/json" -d '{"input": {"action": "action"}}' localhost:8181/v1/data/actions/allow
```

**Comment by migscabral** [comment]
Tried that too, still getting empty response with just the decision id:

Request:
```
curl --location --request POST 'http://localhost:8282/v1/data/actions' \
--header 'Content-Type: application/json' \
--data-raw '{"input": {
    "action": "action",
    "action_group": "action-group-1",
    "role": "role-1"
}}'
```

Reponse:
```
{
    "decision_id": "fbeeaa0b-5e53-49b4-a55a-82909a47f93c"
}
```

**Comment by srenatus** [comment]
```diff
- http://localhost:8282/v1/data/actions/allow
+ http://localhost:8282/v1/data/authz/actions/allow
```

With that, I seem to get the proper responses:

```
$ curl --location --request POST 'http://localhost:8282/v1/data/authz/actions/allow' \
--header 'Content-Type: application/json' \
--data-raw '{"input": {
    "action": "action-1",
    "action_group": "action-group-1",
    "role": "role-1"
}}'
{"decision_id":"6eaa2c8e-b16e-4bd8-8e9e-12ae6375fdda","result":true} 
```

I had played with the authz/actions/policy.rego a bit and changed it slightly:
```rego
package authz.actions

import future.keywords.in

default allow = false

allow {
	input.role == "admin"
}

allow {
	input.action in user_is_granted
}

user_is_granted[grant] {
	some role_action in data.authz.actions.role_actions
	role_action.role == input.role
	some grant in role_action.actions[input.action_group]
}
```
should be equivalent to what you had, anyways.

Hope this helps?

**Comment by anderseknert** [comment]
@migscabral anything left to do here, or are we good to close this one?

**Comment by srenatus** [comment]
🧹 cleanup out dormant issues -- please re-open, or reach out via some other channel if you keep struggling with this.
